File: day3/two.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	fileName := os.Args[1]
	fmt.Printf("%v\n", fileName)
	file, err := os.Open(fileName)
	if err != nil {
		// Handle any errors opening the file
		fmt.Println("Error opening file:", err)
		return
	}
	defer file.Close() // Ensure the file is closed when we're done

	// Create a scanner to read the file
	scanner := bufio.NewScanner(file)

	var sum int64
	sum = 0

	// Read line by line
	for scanner.Scan() {
		line := scanner.Text()
		fmt.Println(line) // Process the line
		arr := strings.Split(line, "")
		// fmt.Println(arr) // Process the line

		numArr := make([]string, 0)

		for index, value := range arr {
			// fmt.Println(index, value)
			if strings.Contains("0123456789", value) {
				numArr = append(numArr, value)
			} else {
				numbers := []string{"one", "two", "three", "four", "five", "six", "seven", "eight", "nine"}

				for nIndex, num := range numbers {
					if index+len(num) <= len(line) && num == line[index:index+len(num)] {
						numArr = append(numArr, strconv.Itoa(nIndex+1))
					}
				}

			}
		}

		// fmt.Printf("%v\n", numArr)
		res, err := strconv.ParseInt(numArr[0]+numArr[len(numArr)-1], 10, 64)

		if err == nil {
			sum += res
		}
		fmt.Printf("%v\n", res)

	}

	fmt.Printf("Sum: %v\n", sum)
	// Check for any errors encountered while reading
	if err := scanner.Err(); err != nil {
		fmt.Println("Error reading from file:", err)
	}
}

File: day4/two.ts
import * as fs from "fs";

function main() {
  const filePath = process.argv[2]; // Replace with the path to your file

  const fileContent = fs.readFileSync(filePath, "utf-8");
  const lines = fileContent.split("\n");

  let sum = 0;

  let names = [
    "one",
    "two",
    "three",
    "four",
    "five",
    "six",
    "seven",
    "eight",
    "nine",
  ];

  for (const line of lines) {
    // Process each line here
    let items = [];
    for (let i = 0; i < line.length; i++) {
      if ("123456789".includes(line[i])) {
        items.push(line[i]);
      }

      for (let j = 0; j < names.length; j++) {
        let word = line.substring(i, i + names[j].length);
        if (word === names[j]) {
          console.log("true", word);
          items.push(`${j + 1}`);
        }
      }
    }

    if (items.length > 0) {
      let num = items[0] + items[items.length - 1];
      console.log("val: ", num, items);
      let val = parseInt(num);
      sum += val;
    }

    console.log("SUM: ", sum);
  }

  console.log("File reading finished.");
}

main();
